Restore and update Payment Services dropin docs (#852)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/containers/credit-card.mdx

  label: CreditCard
---

import TableWrapper from '@components/TableWrapper.astro';

> The `CreditCard` container renders a credit card form that shoppers can use to enter credit card details during the checkout process.

Reviewer: ```suggestion
The `CreditCard` container renders a form for entering credit card details during checkout.
```

<div style="background-color: var(--sl-color-blue-low); border-left: 4px solid var(--sl-color-blue); padding: 0.75rem 1rem; border-radius: 0.25rem; margin: 1rem 0;">
<strong>Version: 3.1.1</strong>
</div>

## Configuration

The `CreditCard` container provides the following configuration options:

<TableWrapper nowrap={[0, 1, 2]}>
| Option | Type | Req? | Description |
|--------|------|------|-------------|

> | `getCartId` | function | Yes | Should return a promise that resolves to the shopper's cart ID. |

Reviewer: ```suggestion
| `getCartId` | function | Yes | Returns a promise that resolves to the shopper’s cart ID. |
```

> | `creditCardFormRef` | object | Yes | Credit card form reference. Initially, `{ current: null }` should be passed. Once rendered, the credit card container will set the `current` property to an `{ validate: () => boolean; submit: () => Promise<void> }` object, which parent containers can use to (programmatically) validate and submit the credit card form. Any error that happens during the payment flow will not only propagate to `onError` but will also cause the promise returned by `submit()` to be rejected. |

Reviewer: ```suggestion
| `creditCardFormRef` | object | Yes | Reference to the credit card form. Initially, `{ current: null }` should be passed. After rendering, the container sets `current` to the `validate: () => boolean` and `submit: () => Promise<void>` object, which parent containers can use to programmatically validate and submit the form. Any error during the payment flow propagates to `onError` and causes the promise returned by `submit()` to be rejected.|
```

Author: We should keep the contract that if the payment flow fails, the promise returned by `submit()` will reject.

This guarantee allows SIs to simplify their integration: if all payment are guaranteed to make the `submit()` promise reject, then SIs are not required to pass in an `onError` callback anymore.

> | `onSuccess` | function | No | Called when the payment flow finishes successfully. The function receives an `{ cartId: string }` object as an argument. |

Reviewer: ```suggestion
| `onSuccess` | function | No | Called when the payment completes successfully. Receives `{ cartId: string }`. |
```

> | `onError` | function | No | Called when the payment flow was aborted due to an error. The function receives an object with two properties: `{name: string, message: string}`, containing the localized error name and message. Both properties are user-facing and can be translated using the `PaymentServices.CreditCard.errors` [language definitions](/dropins/payment-services/dictionary/). |

Reviewer: ```suggestion
| `onError` | function | No | Called when the payment flow fails or is aborted. Receives `{ name: string, message: string }`, containing localized, user-facing error details. These values can be translated using `PaymentServices.CreditCard.errors` [language definitions](/dropins/payment-services/dictionary/). |
```
